src/AgentModel.cpp

```diff
@@ -848,45 +994,52 @@ double AgentModel::subconsciousLateralControl() {
             _state.aux[31] += _state.conscious.lateral.paths[j].factor * _state.aux[idx];
 
         }
+    }
 
+    // reset temporary theta and dTheta
+    if (_memory.laneChange.switchLane != 0) {
+        for (int i = 0; i < agent_model::NOCP * agent_model::NORP * 2; i++) 
+            _state.aux[i] = 0;
     }
 
     return reaction;
-
 }
 
-
 double AgentModel::subconsciousFollow() {
 
     using namespace std;
 
+    double res = 0;
     // get values
-    double vT = _state.conscious.follow.velocity;
-    double ds = _state.conscious.follow.distance;
-    double v0 = _state.conscious.velocity.local;
-    double s0 = _param.follow.dsStopped;
-    double T = _param.follow.timeHeadway;
-    double TMax = _param.follow.thwMax;
-    double v = _input.vehicle.v;
-
-    // ignore when distance is inf
-    if (std::isinf(ds))
-        return 0.0;
-
-    double v0T = std::max(10.0, v0);
-    double vTT = std::min(v0T, std::max(5.0, vT));
+    for (auto &t : _state.conscious.follow.targets) {
+
+        double vT = t.velocity;
+        double ds = t.distance;
+        double v0 = _state.conscious.velocity.local;
+        double s0 = _param.follow.dsStopped;
+        double T = _param.follow.timeHeadway;
+        double TMax = _param.follow.thwMax;
+        double v = _input.vehicle.v;
+
+        // ignore when distance is inf
```

Comment:
we could move this condition to the first line of the loop

Reply:
Fixed